rcl/src/rcl/arguments.c

```diff
@@ -85,7 +85,7 @@ rcl_arguments_get_param_files(
   RCL_CHECK_ARGUMENT_FOR_NULL(arguments->impl, RCL_RET_INVALID_ARGUMENT, allocator);
   RCL_CHECK_ARGUMENT_FOR_NULL(parameter_files, RCL_RET_INVALID_ARGUMENT, allocator);
   *(parameter_files) = allocator.allocate(
-    sizeof(char *) * arguments->impl->num_param_files_args, allocator.state);
+    sizeof(char *) * (size_t)arguments->impl->num_param_files_args, allocator.state);
```

Comment:
Hm, it looks like `num_params_files_args` (https://github.com/ros2/rcl/blob/master/rcl/src/rcl/arguments_impl.h#L38) is always used as an unsigned number (which makes sense, since it is the "number" of things).  Thus, I'd suggest that we actually change the type in the structure, rather than casting it here (in a quick test I did here, that seems to be OK).